# what the hack is MEAN, MEDIAN, MODE

### Mean:
let's say we have array of numbers 

In [11]:
import numpy as np
awesomeArray = np.array([22,23,23,25,26,27,42])
sumOfit = np.sum(awesomeArray)
length = len(awesomeArray)
mean = np.mean(awesomeArray)
median = np.median(awesomeArray)
counts = np.bincount(awesomeArray)
mode = np.argmax(counts)
print("sum: {}".format(sumOfit))
print("length: {}".format(length))
print("mean: {}".format(mean))
print("median: {}".format(median))
print("mode: {}".format(mode))


sum: 188
length: 7
mean: 26.857142857142858
median: 25.0
mode: 23


so basically it's takes sum of the numbers

> 22 + 23 + 23 + 25 + 26 + 27 + 42  = 188

and divides it too length of the array.

> 188 / 7 = 26.8571428571


### Median
similar to mean but with diferent approach.
first it's sorts elements in the array

In [13]:
sortedawesomeArray = np.sort(awesomeArray)
print(sortedawesomeArray)

[22 23 23 25 26 27 42]


takes the element in the center
in this case our median is *25* 
if we had even number it's takes 2 number which closer to center. 
sums them up and divede by 2
let say this was our array
> [22 23 23 25 26 26 27 42]

the number close to center is  25 and 26
> (25 + 26) / 2 = 25.5




### Mode
most repeated element in the array.
